go_lib/hooks/internal_tls/hook.go

```diff
@@ -55,6 +58,23 @@ type GenSelfSignedTLSHookConf struct {
 	// Data in values store as plain text
 	// In helm templates you need use `b64enc` function to encode
 	FullValuesPathPrefix string
+
+	// You can set Paths for internal values explicitly or use FullValuesPathPrefix
+	CAValuesPath   string
```

Comment:
When I did this hook, we discussed with @nabokihms about this paths. I as you added internal paths in the configuration and we decided that is overkill. Better way use `FullValuesPathPrefix` and change paths in module.

Reply:
Well. This is not comfortable because:
a) We are losing a flexibility
b) `FullValuesPathPrefix` gives us a plain structure while the vast majority of our modules have a different structure, like:
```
      customCertificateData:
        type: object
        properties:
          tls.crt:
            type: string
            x-examples:
              - plainstring
          tls.key:
            type: string
            x-examples:
              - plainstring
          ca.crt:
            type: string
            x-examples:
              - plainstring
```
or 
```
          serverCerts:
            type: object
            default: {}
            properties:
              ca:
                type: string
                x-examples: ["---ROOT PEM---"]
              key:
                type: string
                x-examples: ["---KEY PEM---"]
              cert:
                type: string
                x-examples: ["---CERT PEM---"]
```

Simplify them to a plain struct will create more chaos imho, look at cilium-hubble module for example, it has a lot of certs.


That's why I kept backward compatibility but I want to have an ability to set values path explicitly

Reply:
We are waiting @nabokihms  here.

Reply:
But now it's realized in [another way](https://github.com/deckhouse/deckhouse/blob/main/go_lib/hooks/internal_tls/hook.go#L127) and is used like [this](https://github.com/deckhouse/deckhouse/blob/main/modules/301-prometheus-metrics-adapter/openapi/values.yaml#L24).

Your case will look like:
```
a:
  myCertPEM: string
  myCertCA: string
  myCertKey: string
b:
  myCertPEM: string
  myCertCA: string
  myCertKey: string
  anotherCertPEM: string
  anotherCertCA: string
  anotherCertKey: string
```

Reply:
We can change it like this:
```
ca = strings.Join([]string{FullValuesPathPrefix, "ca"}, ".")
key = strings.Join([]string{FullValuesPathPrefix, "key"}, ".")
cert = strings.Join([]string{FullValuesPathPrefix, "cert"}, ".")
```

then it would work like a charm

Reply:
Let's go with your proposal then. Looks great.

Reply:
Maybe we also should trim dots at the end of `FullValuesPathPrefix`.

Reply:
Done